### DISCUSSION TITLE
Include Script with Async tag

### DISCUSSION THREAD

**Discussion by mdmontesinos** [oqtane/oqtane.framework#2354] [Q&A] [answered]
I'm trying to create a module with a Programmable Search Engine (the one from Google), but I believe I'm facing the issue of not being able to register a script with the 'async' tag. 

Google tutorials state that you should add the following code to the body of where the search engine should be displayed:
```html
<script async src="https://cse.google.com/cse.js?cx=d64cd082822024b16"></script>
<div class="gcse-search"></div>`
```

Right now, I just placed the div in my module, and registered the script as a resource:
```c#
public override List<Resource> Resources => new List<Resource>()
    {
        new Resource { ResourceType = ResourceType.Stylesheet, Url = ModulePath() + "Module.css" },
        new Resource { ResourceType = ResourceType.Script, Url = ModulePath() + "Module.js" },
        new Resource { ResourceType = ResourceType.Script, Location = ResourceLocation.Body, Url = "https://cse.google.com/cse.js?cx=My-Id" }
    };
```
Initially, it seems to work fine, because I access the page and it appears. However, if I go to another page and then come back, the search engine does not appear anymore, and I have to refresh it. 

¿Could this have anything to do with the 'async' tag of the script?
 
I've also tried using a boolean variable to ensure the div is not displayed until the component has initialized.

**Comment by sbwalker** [comment, marked as answer]
I do not believe this has anything to do with async. It is more likely that this a JS Interop issue. You are obviously registering the script correctly but it is not clear to me how the script is being executed as you have not mentioned anything about JS Interop methods. Even if the script is intended to run "onload" you usually need a JS Interop method to call it. This is not an Oqtane requirement but rather a requirement of Blazor.

**Reply by mdmontesinos** [reply]
You were right about the issue being related with JS Interop and I managed to get it working.

Thanks a lot for pointing me in the right direction!

**Reply by sbwalker** [reply]
You're welcome! JavaScript integration is one of the more challenging areas in Blazor.
